internal/log/journald.go

```diff
@@ -42,7 +46,19 @@ func fields(e *apexLog.Entry) map[string]string {
 func (h *journalHandler) HandleLog(e *apexLog.Entry) error {
 	msg := e.Message
 	pri := priority(e)
-	fields := fields(e)
+	fields := journalFields(e)
+
+	if h.test {
+		mu := &sync.Mutex{}
```

Comment:
This part doesn't seem to make sense because a new mutex is created every time - for synchronization to occur, the mutex has to be shared.

If you make the mutex belong to journalHandler it should work.